var/spack/repos/builtin/packages/py-tensorboard/package.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+class PyTensorboard(Package):
+    """TensorBoard is a suite of web applications for
+    inspecting and understanding your TensorFlow runs and
+    graphs."""
+
+    homepage = "https://pypi.python.org/project/tensorboard"
+    url      = "https://github.com/tensorflow/tensorboard/archive/2.2.0.tar.gz"
+
+    version('2.2.0', sha256='d0dfbf0e4b3b5ebbc3fafa6d281d4b9aa5478eac6bac3330652ab6674278ab77')
+
+    depends_on('python@2.7:2.8,3.2:', type=('build', 'run'))
+    depends_on('bazel@0.26.1:', type='build')
+    depends_on('py-setuptools@41.0.0:', type='build')
+    depends_on('py-absl-py@0.4:', type=('build', 'run'))
+    depends_on('py-markdown@2.6.8:', type=('build', 'run'))
+    depends_on('py-requests@2.21.0:3.0.0', type=('build', 'run'))
+    depends_on('py-futures@3.1.1:', type=('build', 'run'), when='^python@:2')
+    depends_on('py-grpcio@1.23.3:', type=('build', 'run'))
+    depends_on('py-google-auth@1.6.3:1.99.99', type=('build', 'run'))
+    depends_on('py-numpy@1.12.0:', type=('build', 'run'))
+    depends_on('py-protobuf@3.6.0:', type=('build', 'run'))
+    depends_on('py-six@1.10.0:', type=('build', 'run'))
+    depends_on('py-werkzeug@0.11.15:', type=('build', 'run'))
+    depends_on('py-wheel', type=('build', 'run'))
+    depends_on('py-wheel@0.26:', type=('build', 'run'), when='@0.6: ^python@3:')
+    depends_on('py-google-auth-oauthlib@0.4.1:0.5', type=('build', 'run'))
+    depends_on('py-tensorboard-plugin-wit', type=('build', 'run'), when='@2.2.0:')
+    depends_on('py-tensorflow-estimator', type='run')
+
+    extends('python')
+
+    phases = ['configure', 'build', 'install']
+
+    def setup_build_environment(self, env):
+        tmp_path = '/tmp/spack/tb'
```

Comment:
I wonder if we could use the `tempfile` builtin library to generate this. Not all systems will have a `/tmp` directory. Not required for this PR, but something to think about for our Bazel packages.

Reply:
yes, not to mention the cleanup aspect - I've unintentionally filled /tmp multiple times with these builds.